libsolidity/ast/Types.cpp

```diff
@@ -381,25 +450,12 @@ MemberList::MemberMap Type::boundFunctions(Type const& _type, ASTNode const& _sc
 				members.emplace_back(&_function, asBoundFunction, *_name);
 	};
 
-	for (UsingForDirective const* ufd: usingForDirectives)
-	{
-		// Convert both types to pointers for comparison to see if the `using for`
-		// directive applies.
-		// Further down, we check more detailed for each function if `_type` is
-		// convertible to the function parameter type.
-		if (
-			ufd->typeName() &&
-			*TypeProvider::withLocationIfReference(typeLocation, &_type, true) !=
-			*TypeProvider::withLocationIfReference(
-				typeLocation,
-				ufd->typeName()->annotation().type,
-				true
-			)
-		)
-			continue;
-
-		for (auto const& pathPointer: ufd->functionsOrLibrary())
+	for (UsingForDirective const* ufd: usingForDirectivesForType(_type, _scope))
+		for (auto const& [pathPointer, operator_]: ufd->functionsAndOperators())
 		{
+			if (operator_)
+				continue;
```

Comment:
This could use a comment:
```suggestion
			if (operator_)
				// Functions used to define operators are not bound to the type.
				// I.e. `using {f} for T` allows `T x; x.f()` but `using {f as +} for T` does not.
				continue;
```

Reply:
comment added